allow to make comments for non-abstract models by COMMENTS_FOR_CONCRETE_MODEL setting

PetrDlouhy opened the pull request.

PetrDlouhy commented:
@claudep I just bumped into this and realized that I forgot to finish it.
Now I have written the tests.
How can I let them run here on GitHub?

claudep commented:
I would love to get a quick second review by someone else (maybe @felixxm ?). I think some documentation may still be needed.

claudep commented on tests/testapp/models.py:
```diff
@@ -15,13 +16,29 @@ def __str__(self):
     headline = models.CharField(max_length=100)
 
+    def save(self, *args, **kwargs):
+        import pudb; pudb.set_trace()
```
oops :-)

PetrDlouhy replied:
Sorry :-D Fixed.

PetrDlouhy commented:
I fixed the useless code and added some basic documentation.

felixxm commented on django_comments/forms.py:
```diff
@@ -137,8 +137,12 @@ def get_comment_create_data(self, site_id=None):
+            content_type=ContentType.objects.get_for_model(
+                self.target_object,
+                for_concrete_model=COMMENTS_FOR_CONCRETE_MODEL,
+            ),
```
I'm not sure if the new setting is the preferred way :thinking: What do you think about adding the `for_concrete_model=True` keyword argument to the `get_comment_object()` and `get_comment_create_data()`?

PetrDlouhy commented:
@felixxm I have rewritten the code to use the `for_concrete_model` parameter.
I had to use `try-catch` to handle overrides that doesn't count with the extra parameter such as: https://github.com/danirus/django-comments-xtd/blob/2ef74e3127f669293504fe88f46480456cca1809/django_comments_xtd/forms.py#L59

I will probably use the COMMENTS_FOR_CONCRETE_MODEL for `django-comments-xtd` implementation anyway, because there it has to be on several places across the application and I can't think of better way.

Another way to support this would be to have some mapping in settings like `COMMENTS_ALLOWED_PROXY_MODELS` which would allow to have different settings for every model.
But I don't think that would be necessary - I think that having `for_concrete_model=False` everywhere would be good default except it would change behavior for current implementations.

PetrDlouhy commented:
I have implemented functionality that I need in #188, however this might be helpful in different use cases so I converted it to draft. 
Anybody feel free to take this over.

blockchainGuru1018 commented:
It's looks good now.

PetrDlouhy commented:
@blockchainGuru1018 Could you please rather review #188. I converted this to draft in case someone wants to tak this work over and adapt it for his/hers purposes. Overridable model keys are solving my use-case much better.

blockchainGuru1018 commented:
> @blockchainGuru1018 Could you please rather review #188. I converted this to draft in case someone wants to tak this work over and adapt it for his/hers purposes. Overridable model keys are solving my use-case much better.

Reviewed your PR @PetrDlouhy
